src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -77,22 +82,41 @@ object DataTypes {
     @JsonScalaEnumeration(classOf[OutputTypeRef]) output: Option[OutputType]
   ) {
     /** Method which validates query fields, as jackson runs on top of runtime reflection so NPE can happen if fields are missing */
-    def validate(entity: String): Either[List[QueryValidationError], Query] = {
+    def validate(entity: String, tezosPlatformDiscovery: TezosPlatformDiscoveryOperations)(implicit ec: ExecutionContext):
+    Future[Either[List[QueryValidationError], Query]] = {
+      import cats.implicits._
+
       val query = Query().patchWith(this)
 
       val invalidQueryFields = query
         .fields
-        .filterNot(field => TezosPlatformDiscoveryOperations.areFieldsValid(entity, Set(field)))
-        .map(InvalidQueryField)
+        .map(field => tezosPlatformDiscovery.areFieldsValid(entity, Set(field)).map(_ -> field))
+        .sequence
+        .map(_.filterNot(_._1).map(fieldPair => InvalidQueryField(fieldPair._2)))
```

Comment:
Please, use named parameters instead of expressions like ``_._1``. It's hard to realize what actually happens in this code.

Reply:
ok, changed